Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -927,6 +927,30 @@ public final class PackageBuilder {
                 case .linker:
                     decl = .OTHER_LDFLAGS
                 }
+
+            case .futureFeatures(let _values):
+                switch setting.tool {
+                case .c, .cxx, .linker:
+                    throw InternalError("only Swift supports future features")
+
+                case .swift:
+                    decl = .OTHER_SWIFT_FLAGS
+                }
+
+                values = _values.precedeElements(with: "-enable-future-feature")
+
+            case .experimentalFeatures(let _values):
+                switch setting.tool {
+                case .c, .cxx, .linker:
+                    throw InternalError(
+                        "only Swift supports experimental features")
+
+                case .swift:
+                    decl = .OTHER_SWIFT_FLAGS
+                }
+
+                values = _values.precedeElements(
+                    with: "-enable-experimental-feature")
```

Comment:
I don't think `precedeElements(with:)` is needed, if the new APIs are implemented like the existing `.define` API, which only stores a single value.

```swift
case .enableExperimentalFeature(let value):
    values = ["-enable-experimental-feature", value]
    switch setting.tool {
```